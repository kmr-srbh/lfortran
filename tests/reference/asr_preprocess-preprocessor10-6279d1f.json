{
    "basename": "asr_preprocess-preprocessor10-6279d1f",
    "cmd": "lfortran --cpp --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/preprocessor10.f90",
    "infile_hash": "128670b190f1e59f40192f15434c3b8d7a85c15fbc901a72e7f7ebc3",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr_preprocess-preprocessor10-6279d1f.stdout",
    "stdout_hash": "a603c0f050de9f0395551e6183fc6af2d37b1d6b7a7ebe99bc3b6573",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}