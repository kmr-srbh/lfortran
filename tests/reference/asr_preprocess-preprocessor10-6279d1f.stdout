(TranslationUnit
    (SymbolTable
        1
        {
            preprocessor10:
                (Program
                    (SymbolTable
                        2
                        {
                            
                        })
                    preprocessor10
                    []
                    [(Print
                        [(StringConstant
                            "X is defined"
                            (Character 1 12 () PointerString)
                        )]
                        ()
                        ()
                    )
                    (Print
                        [(StringConstant
                            "Y is defined"
                            (Character 1 12 () PointerString)
                        )]
                        ()
                        ()
                    )
                    (Print
                        [(StringConstant
                            "Z is not defined"
                            (Character 1 16 () PointerString)
                        )]
                        ()
                        ()
                    )
                    (Print
                        [(StringConstant
                            "X and Y is defined"
                            (Character 1 18 () PointerString)
                        )]
                        ()
                        ()
                    )
                    (Print
                        [(StringConstant
                            "Z and Y is not defined"
                            (Character 1 22 () PointerString)
                        )]
                        ()
                        ()
                    )
                    (Print
                        [(StringConstant
                            "Z or Y is defined"
                            (Character 1 17 () PointerString)
                        )]
                        ()
                        ()
                    )
                    (Print
                        [(StringConstant
                            "X and Y and Z is not defined"
                            (Character 1 28 () PointerString)
                        )]
                        ()
                        ()
                    )
                    (Print
                        [(StringConstant
                            "X and Y or Z is defined"
                            (Character 1 23 () PointerString)
                        )]
                        ()
                        ()
                    )
                    (Print
                        [(StringConstant
                            "X and Y or Z is defined"
                            (Character 1 23 () PointerString)
                        )]
                        ()
                        ()
                    )
                    (Print
                        [(StringConstant
                            "X and !Y or !Z is defined"
                            (Character 1 25 () PointerString)
                        )]
                        ()
                        ()
                    )
                    (Print
                        [(StringConstant
                            "X and !(Y or Z) is not defined"
                            (Character 1 30 () PointerString)
                        )]
                        ()
                        ()
                    )]
                )
        })
    []
)
